qa/tasks/ceph_manager.py

```diff
@@ -659,27 +659,97 @@ def fix_pgp_num(self, pool=None):
 
     def test_pool_min_size(self):
         """
-        Kill and revive all osds except one.
+        Loop to selectively push PGs below their min_size and test that recovery
+        still occurs.
         """
         self.log("test_pool_min_size")
         self.all_up()
         self.ceph_manager.wait_for_recovery(
             timeout=self.config.get('timeout')
             )
-        the_one = random.choice(self.in_osds)
-        self.log("Killing everyone but %s", the_one)
-        to_kill = filter(lambda x: x != the_one, self.in_osds)
-        [self.kill_osd(i) for i in to_kill]
-        [self.out_osd(i) for i in to_kill]
-        time.sleep(self.config.get("test_pool_min_size_time", 10))
-        self.log("Killing %s" % (the_one,))
-        self.kill_osd(the_one)
-        self.out_osd(the_one)
-        self.log("Reviving everyone but %s" % (the_one,))
-        [self.revive_osd(i) for i in to_kill]
-        [self.in_osd(i) for i in to_kill]
-        self.log("Revived everyone but %s" % (the_one,))
-        self.log("Waiting for clean")
+
+        minout = int(self.config.get("min_out", 1))
+        minlive = int(self.config.get("min_live", 2))
+        mindead = int(self.config.get("min_dead", 1))
+        self.log("doing min_size thrashing")
+        assert self.ceph_manager.is_clean(), \
+            'not clean before minsize thrashing starts'
+        while not self.stopping:
+            # look up k and m from all the pools on each loop, in case it
+            # changes as the cluster runs
+            k = 0
+            m = 99
+            has_pools = False
+            pools_json = self.get_osd_dump_json()['pools']
```

Comment:
```
, u'epoch': 1, u'modified': u'0.000000'}, u'fsid': u'9f9bb9d1-8011-4afc-bc3a-237e70dc3946', u'quorum': [0, 1, 2]}
2019-04-03T07:55:22.282 INFO:tasks.thrashosds.thrasher:Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2019-04-01-1531/qa/tasks/ceph_manager.py", line 953, in wrapper
    return func(self)
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2019-04-01-1531/qa/tasks/ceph_manager.py", line 1083, in do_thrash
    self.choose_action()()
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2019-04-01-1531/qa/tasks/ceph_manager.py", line 703, in test_pool_min_size
    pools_json = self.get_osd_dump_json()['pools']
AttributeError: Thrasher instance has no attribute 'get_osd_dump_json'
```

http://pulpito.ceph.com/kchai-2019-04-03_07:35:34-rados-wip-kefu-testing-2019-04-01-1531-distro-basic-smithi/3804803

`get_osd_dump_json()` is a method of `CephManager` not `Thrasher`.